cmd/tuf/remove_key.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/theupdateframework/go-tuf"
+	"github.com/flynn/go-docopt"
+)
+
+func init() {
+	register("remove-key", cmdRemoveKey, `
+usage: tuf remove-key [--expires=<days>] <role> <id>
+
+Remove a signing key
+
+Before the key is removed the key will be first revoked
+The key will then be removed from the root metadata file and if the key is present in 
+"keys" directory it will also be removed
+
+Options:
+  --expires=<days>   Set the root metadata file to expire <days> days from now.
+`)
+}
+
+func cmdRemoveKey(args *docopt.Args, repo *tuf.Repo) error {
+	role := args.String["<roles>"]
+	keyID := args.String["<id>"]
+
+	// Extract the --ecpires option value
+	expires, _ := args.Int("--expires")
+
+	// Revoke the key 
+	if err := repo.RevokeKey(role, keyID); err != nil {
+		return err
+	}
+
+	// Update expiration vaalue if --expires is provided 
+	if expires > 0 {
+		rooMetadata := repo.Root()
+		if rootMetadata != nil {
+			newExpires := time.Now().Add(time.Duration(expires) * 24 * time.Hour)
+			rootMetadata.Expires = &newExpires 
+			if err := repo.Commit(); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
I don't think there's need to do all of that, for example see how the revoke key command is implemented and copy that part from it - https://github.com/theupdateframework/go-tuf/blob/master/cmd/tuf/revoke_key.go 👍

Reply:
would it be better to call the revoke_key command in the remove_key command? instead of having to copy it over?

Reply:
It should work fine, yes 👍

Reply:
Great! so then what i did was call the cmdRevokeKey(args, repo) in the remove_key command then you can take into consideration the expiration value if present. Hopefully that should work.